src/main/kotlin/com/dansplugins/factionsystem/command/faction/addmember/MfFactionAddMemberCommand.kt

```diff
@@ -0,0 +1,99 @@
+package com.dansplugins.factionsystem.command.faction.addmember
+
+import com.dansplugins.factionsystem.MedievalFactions
+import com.dansplugins.factionsystem.faction.MfFactionMember
+import com.dansplugins.factionsystem.player.MfPlayer
+import dev.forkhandles.result4k.onFailure
+import org.bukkit.ChatColor
+import org.bukkit.command.Command
+import org.bukkit.command.CommandExecutor
+import org.bukkit.command.CommandSender
+import org.bukkit.command.TabCompleter
+import org.bukkit.entity.Player
+import preponderous.ponder.command.dropFirst
+import java.util.logging.Level
+
+class MfFactionAddMemberCommand(private val plugin: MedievalFactions) : CommandExecutor, TabCompleter {
+
+    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
+        if (!sender.hasPermission("mf.force.addmember")) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberNoPermission"]}")
+            return true
+        }
+        if (sender !is Player) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberNotAPlayer"]}")
+            return true
+        }
+        if (args.isEmpty()) {
+            sender.sendMessage("${ChatColor.RED}${plugin.language["CommandFactionAddMemberUsage"]}")
+            return true
+        }
+
+        // get target player
+        val targetPlayer = plugin.server.getOfflinePlayer(args[0])
```

Comment:
I think we want the faction to be specified as well in this command, so we can add members to any other faction?

Reply:
The faction is specified on line 56.

Reply:
Ah I see! Maybe it would be better to have the faction as the first argument and use the last argument for player name?

Reply:
I view it almost like a sentence: Add <player> to <faction>

Also, if we switch the arguments around then we won't be able to easily join the arguments to create the faction name unless we remove the last argument first.

What do you think?

Reply:
If it makes sense to you that's fine.
My general approach is to put things that vary the least first but that's definitely not a hard-and-fast rule.